crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -297,14 +295,25 @@ impl SettlementClient for EthereumSettlementClient {
 mod test_config {
     use super::*;
     use alloy::network::TransactionBuilder;
-    use tests::{ADDRESS_TO_IMPERSONATE, SHOULD_IMPERSONATE_ACCOUNT, TEST_NONCE};
 
-    pub fn configure_transaction(tx_envelope: TxEnvelope) -> TransactionRequest {
+    pub async fn configure_transaction(
+        provider: Arc<RootProvider<Http<Client>>>,
+        tx_envelope: TxEnvelope,
+        impersonate_account: Option<Address>,
+    ) -> TransactionRequest {
         let mut txn_request: TransactionRequest = tx_envelope.into();
 
-        if *SHOULD_IMPERSONATE_ACCOUNT {
-            txn_request.set_nonce(*TEST_NONCE);
-            txn_request = txn_request.with_from(*ADDRESS_TO_IMPERSONATE);
+        // IMPORTANT to understand #[cfg(test)], #[cfg(not(test))] and SHOULD_IMPERSONATE_ACCOUNT
+        // Two tests :  `update_state_blob_with_dummy_contract_works` & `update_state_blob_with_impersonation_works` use a env var `SHOULD_IMPERSONATE_ACCOUNT` to inform the function `update_state_with_blobs` about the kind of testing,
+        // `SHOULD_IMPERSONATE_ACCOUNT` can have any of "0" or "1" value :
+        //      - if "0" then : Testing via default Anvil address.
+        //      - if "1" then : Testing via impersonating `Starknet Operator Address`.
+        // Note : changing between "0" and "1" is handled automatically by each test function, `no` manual change in `env.test` is needed.
+        if let Some(impersonate_account) = impersonate_account {
+            let nonce =
+                provider.get_transaction_count(impersonate_account).await.unwrap().to_string().parse::<u64>().unwrap();
```

Comment:
We already send the `nonce` to the `update_state_with_blobs` method then why are we recalculating it here ?